lib/model-builder.js

```diff
@@ -17,7 +17,8 @@ var deprecated = require('depd')('loopback-datasource-juggler');
 var DefaultModelBaseClass = require('./model.js');
 var List = require('./list.js');
 var ModelDefinition = require('./model-definition.js');
-var mergeSettings = require('./utils').mergeSettings;
+var deepMerge = require('./utils').deepMerge;
```

Comment:
`mergeSettings` is not a good term, as
- it is also used for properties
- it is now misleading in the context of the new merge policy

I propose to use its alias `deepMerge` (already defined in utils.js)